packages/mds-geography/api.ts

```diff
@@ -0,0 +1,185 @@
+    checkAccess(scopes => scopes.includes('policies:write')),
+    async (req, res) => {
+      const geography = req.body
+      const details = geographyValidationDetails(geography)
+
+      if (details != null) {
+        await log.error('invalid policy json', details)
+        return res.status(400).send(details)
+      }
+
+      try {
+        await db.editGeography(geography)
+        return res.status(201).send(geography)
+      } catch (err) {
+        await log.error('failed to write geography', err.stack)
+        return res.status(404).send({ result: 'not found' })
+      }
+    }
+  )
+
+  app.delete(
+    pathsFor('/geographies/:geography_id'),
+    checkAccess(scopes => scopes.includes('policies:delete')),
+    async (req, res) => {
+      const { geography_id } = req.params
+      try {
+        await db.deleteGeography(geography_id)
+        return res.status(200).send({ result: `Successfully deleted geography of id ${geography_id}` })
+      } catch (err) {
+        await log.error('failed to delete geography', err.stack)
+        return res.status(404).send({ result: 'geography either not found or already published' })
+      }
+    }
+  )
+
+  app.get(
+    pathsFor('/geographies/:geography_id/meta'),
+    checkAccess(scopes => scopes.includes('policies:read')),
+    async (req, res) => {
+      const { geography_id } = req.params
+      try {
+        const geography_metadata = await db.readSingleGeographyMetadata(geography_id)
+        return res.status(200).send(geography_metadata)
+      } catch (err) {
+        await log.error('failed to read geography metadata', err.stack)
+        return res.status(404).send({ result: 'not found' })
+      }
+    }
+  )
+
+  app.get(
+    pathsFor('/geographies'),
+    checkAccess(scopes => scopes.includes('policies:read')),
+    async (req, res) => {
+      const summary = req.query.summary === 'true'
+      try {
+        const geographies = summary ? await db.readGeographySummaries() : await db.readGeographies()
+        return res.status(200).send(geographies)
+      } catch (err) {
+        await log.error('failed to read geographies', err.stack)
+        return res.status(404).send({ result: 'not found' })
+      }
+    }
+  )
+
+  app.put(
+    pathsFor('/geographies/:geography_id/meta'),
+    checkAccess(scopes => scopes.includes('policies:write')),
+    async (req, res) => {
+      const geography_metadata = req.body
+      try {
+        await db.updateGeographyMetadata(geography_metadata)
+        return res.status(200).send(geography_metadata)
+      } catch (updateErr) {
+        if (updateErr instanceof NotFoundError) {
+          try {
+            await db.writeGeographyMetadata(geography_metadata)
+            return res.status(201).send(geography_metadata)
+          } catch (writeErr) {
+            await log.error('failed to write geography metadata', writeErr.stack)
+            return res.status(500).send(new ServerError())
```

Comment:
Need better error handling here, also don't think we want geo metadata for a geo that doesn't exist

Reply:
Oof you're right.